2996_reflection/reflection.md

```diff
@@ -4305,27 +4403,79 @@ namespace std::meta {
+<tr><td>[`op_less`]{.addu}</td><td>[`operator<`]{.addu}</td></tr>
+<tr><td>[`op_greater`]{.addu}</td><td>[`operator>`]{.addu}</td></tr>
+<tr><td>[`op_less_equals`]{.addu}</td><td>[`operator<=`]{.addu}</td></tr>
+<tr><td>[`op_greater_equals`]{.addu}</td><td>[`operator>=`]{.addu}</td></tr>
+<tr><td>[`op_three_way_compare`]{.addu}</td><td>[`operator<=>`]{.addu}</td></tr>
+<tr><td>[`op_ampersand_ampersand`]{.addu}</td><td>[`operator&&`]{.addu}</td></tr>
+<tr><td>[`op_pipe_pipe`]{.addu}</td><td>[`operator||`]{.addu}</td></tr>
+<tr><td>[`op_less_less`]{.addu}</td><td>[`operator<<`]{.addu}</td></tr>
+<tr><td>[`op_greater_greater`]{.addu}</td><td>[`operator>>`]{.addu}</td></tr>
+<tr><td>[`op_less_less_equals`]{.addu}</td><td>[`operator<<=`]{.addu}</td></tr>
+<tr><td>[`op_greater_greater_equals`]{.addu}</td><td>[`operator>>=`]{.addu}</td></tr>
+<tr><td>[`op_plus_plus`]{.addu}</td><td>[`operator++`]{.addu}</td></tr>
+<tr><td>[`op_minus_minus`]{.addu}</td><td>[`operator--`]{.addu}</td></tr>
+<tr><td>[`op_comma`]{.addu}</td><td>[`operator,`]{.addu}</td></tr>
+</table>
+
+:::
+:::
 
-[#]{.pnum} *Returns*: The unqualified name of the entity represented by `r`. If this entity has no name, then an empty `string_view` or `u8string_view`, respectively.
+### [meta.reflection.names] Reflection names and locations {-}
 
+::: std
+::: addu
 ```cpp
-consteval string_view qualified_name_of(info r);
-consteval u8string_view u8qualified_name_of(info r);
+consteval string_view identifier_of(info r);
+consteval u8string_view u8identifier_of(info r);
 ```
 
-[#]{.pnum} *Constant When*: If returning `string_view`, the qualified name is representable using the ordinary string literal encoding. `r` represents a declared entity.
+[#]{.pnum} Let *E* be UTF-8 if returning a `u8string_view`, and otherwise the ordinary string literal encoding.
 
-[#]{.pnum} *Returns*: The qualified name of the entity represented by `r`. If this entity has no name, then an empty `string_view` or `u8string_view`, respectively.
+[#]{.pnum} *Constant When*: If `r` represents a function, then when the function is not a constructor, destructor, operator function, or conversion function. Otherwise, if `r` is a function template, then when the function template is not a constructor template, a conversion function template, or an operator function template. Otherwise, if `r` represents a variable, an entity that is not a function or function template, or an alias of a type or namespace, then when the declaration of what is represented by `r` introduces an identifier representable by *E*. Otherwise, if `r` represents a base class specifier for which the base class is a named type, then when the name of that type is an identifier representable by *E*. Otherwise, when `r` represents a description of the declaration of a non-static data member, and the declaration of any data member having the properties represented by `r` would introduce an identifier representable by *E*.
```

Comment:
Ok so this:
```suggestion
[#]{.pnum} *Constant When*: `has_identifier(r)` is `true`.
```
Is very nearly correct. It just doesn't cover the representation part. Which makes me wonder if we need `has_{u8,}identifier` to ensure that we can fully check this?

Reply:
It'd be _pretty_ weird if I had some variable named with a UCN, and `has_identifier` returned false.